### PR TITLE
[clang-tidy] fix performance-* warnings

### PR DESCRIPTION
**Pull request by Rechi**
## Description
fix performance-* clang-tidy warnings

## Types of change
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp [hunk 1]
```diff
@@ -204,7 +204,7 @@ std::unique_ptr<ISavestate> CSavestateDatabase::RenameSavestate(const std::strin
 
   newSavestate->Finalize();
 
-  std::string path = savestatePath;
+  const std::string& path = savestatePath;
   if (!AddSavestate(path, "", *newSavestate))
```
Comment (neo1973) [inline xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp:208]: We can get rid of the `path` variable entirely, right?
```suggestion
  if (!AddSavestate(savestatePath, "", *newSavestate))
```

Comment (garbear) [reply xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp:209]: I believe we can.

Comment (Rechi) [reply xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp]: I've removed the variable. Usually I only apply the fixes offered by clang-tidy.

Comment (garbear) [reply xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp]: It's good that @neo1973 caught this then!

**Review by garbear — APPROVED** [review]
Game OSD changes look good

**Pull request merged**